internal/convert/service.go

```diff
@@ -149,17 +149,33 @@ func toServices(s latest.ServiceConfig, objectMeta metav1.ObjectMeta, labelSelec
 }
 
 // toHeadlessService creates a Kubernetes headless service.
-func toHeadlessService(objectMeta metav1.ObjectMeta, labelSelector map[string]string, original apiv1.Service) *apiv1.Service {
+func toHeadlessService(objectMeta metav1.ObjectMeta, labelSelector map[string]string, original apiv1.Service, internalPorts []latest.InternalPort) *apiv1.Service {
```

Comment:
We should rename this to something like `toInternalService` as it's not necessarily a `HeadlessService` any more.